torch/_subclasses/fake_tensor.py

```diff
@@ -1726,9 +1752,9 @@ def _make_cache_entry(
             if non_cacheable:
                 raise _BypassDispatchCache(f"unbacked symbol in HOP {func} output")
 
-        if output is None:
+        if isinstance(output, (int, torch.SymInt, type(None))):
```

Comment:
@anijain2305 how did you end up seeing an int as an output from invoke_subgraph? I wouldn't have expected this to happen since speculate_subgraph ensures that the outputs are just Tensors or symints

Reply:
just leaving my understanding here - I figured that this PR was more general than just invoke_subgraph, and is meant to get faketensor caching support for `_scaled_dot_product_flash_attention` in general. It returns two SymInt outputs (`max_q`,`max_k`) that can be plain ints when not using dynamic shapes: https://github.com/pytorch/pytorch/blob/main/aten/src/ATen/native/native_functions.yaml#L14912

Reply:
Should we be caching max_q and max_k?

Reply:
they come directly from the input (query/key) tensors' shape: https://github.com/pytorch/pytorch/blob/main/torch/_meta_registrations.py#L5577, so the output SymInt is uniquely determined by input symbolic shapes. As long as fake tensor caching already supports hashing on symbolic shape inputs, then I would imagine that this is safe?

Reply:
What Brian said. Earlier, I was supporting op by op, and I saw ints with `_scaled_dot_product_flash_attention`.

But then I added support for both `symint` and int because I saw `symint` with invoke_subgraph.

Hashing on symbols is effectively turned off today because there was a bug, but Aaron says that he no longer sees the bug and will send a PR to enable it shortly.

Reply:
Nevermind, with static shapes FakeTensor should error out on said custom operator. We're good